### DISCUSSION TITLE
Is it possible to configure Lombok in lunarvim?

### DISCUSSION THREAD

**Discussion by igson** [LunarVim/LunarVim#3910] [General]
Is there any way to configure the java lombok library in LunaVim?

**Comment by igorsyrbu** [comment]
I'd also really appreciate it if someone could provide an installation guide!

**Reply by BtAbdul** [reply]
Here is full guide: https://medium.com/@chrisatmachine/lunarvim-as-a-java-ide-da65c4a77fb4

**Comment by stdcout42** [comment]
Did you get anywhere with this @igson ?

**Comment by gonpra** [comment]
You need to overwrite the LSP execution command to add the lombok jar
Install JDTLS LSP with
`:MasonInstall jdtls`

Then in your configuration file (probably at **$HOME/.config/lvim**) add
`local lspconfig = require('lspconfig')`
`local lombok = vim.fn.stdpath('data') .. '/mason/packages/jdtls/lombok.jar'`

`lspconfig.jdtls.setup {
  cmd = {
    "jdtls",
    "--jvm-arg=-javaagent:" .. lombok
  }
}`

When you run :MasonInstall it also comes with lombok.jar so there shouldn't be any need to download any external files.

**Comment by BtAbdul** [comment]
> 

Yes, you can achieve it by following the instructions by Chris@machine: https://medium.com/@chrisatmachine/lunarvim-as-a-java-ide-da65c4a77fb4
